posts/2020-09-29-microprofile-openapi-beta-200011.adoc

```diff
@@ -0,0 +1,253 @@
+
+This means that you can now try out our in-development Open Liberty features by just adding the relevant coordinates to your build tools (either the All Beta Features package or the Jakarta EE 9 Beta Features package, depending on what you want to try).
+
+If you try either package, <<feedback, let us know what you think>>.
+
+[#allbeta]
+== All Beta Features package
+
+
+image::img/blog/logos/home_logo_microprofile.png[MicroProfile Logo,width=70%,align="center"]
+
+
+The All Beta Features package includes <<openAPI, MicroProfile openAPI 2.0>>
+
+[#openAPI]
+=== MicroProfile OpenAPI 2.0 
+
+MicroProfile OpenAPI 2.0 builds on top of the OpenAPI v3 specification. The link:https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.0.md[OpenAPI v3 specification] defines a standard, language-agnostic, interface for describing REST APIs which allows documentation to be generated from the APIs themselves. The link:https://download.eclipse.org/microprofile/microprofile-open-api-2.0-RC3/microprofile-openapi-spec.html[MicroProfile OpenAPI specification] provides a unified Java API for the OpenAPI v3 specification which allows Java developers to generate OpenAPI v3 documents from their JAX-RS applications.
+MicroProfile OpenAPI 2.0 introduces some new annotations that simplify the process of generating OpenAPI documentation for your REST APIs. It also introduces a new MicroProfile Config property prefix that can be used to define the schema for Java classes that you do not have the source code for.
+
+[#schemaproperty]
+==== @SchemaProperty
+
+The @SchemaProperty annotation has been added to allow you to include properties in the definition of a schema when using the @Schema annotation. Previously, the @Schema annotation did not allow you to define any properties. Properties were introspected from the implementation class.
+With the introduction of the @SchemaProperty annotation, a list of properties can now defined using the properties attribute on the @Schema annotation, as follows:
+
+[source, java]
+----
+@Schema(
+    name = "User",
+        description = "Telephone number to contact the user”,
+        implementation = User.class,
+        properties = {
+            @SchemaProperty(
+                name = "phone",
+                description = "Telephone number to contact the user”
+            )
+        }
+    )} 
+----
+[#requestbodyschema]
+==== @RequestBodySchema
+
+The @RequestBodySchema annotation has been added to provide a shorthand mechanism to specify the schema for a request body. Previously, defining the schema for a request body required the use of three different annotations, as follows:
+[source, java]
+----
+@RequestBody(
+    content = {
+        @Content(
+            schema = @Schema(
+                implementation = MyRequestObject.class
+            )
+        )
+    }
+)
+----
+
+With the introduction of the @RequestBodySchema annotation, this can be simplified to the following:
```

Comment:
"to the following" -> "to the following single annotation"

Reply:
Done